addons/stock/models/stock_quant.py

```diff
@@ -378,6 +478,30 @@ def _onchange_serial_number(self):
             if message:
                 return {'warning': {'title': _('Warning'), 'message': message}}
 
+    def _apply_inventory(self):
+        move_vals = []
+        for quant in self:
+            # Create and validate a move so that the quant matches its `inventory_quantity`.
+            if float_compare(quant.inventory_diff_quantity, 0, precision_rounding=quant.product_uom_id.rounding) > 0:
+                move_vals.append(
+                    quant._get_inventory_move_values(quant.inventory_diff_quantity,
+                                                     quant.product_id.with_company(quant.company_id).property_stock_inventory,
```

Comment:
on large database, maybe prefetch property_stock_inventory by product and company before.

Reply:
The ORM should do the job because of the `_prefetch_ids` of the mapped of `quant.product_id` contains `self.product_id.ids`.

Reply:
Prefetch should be working